artifactory/commands/golang/go.go

```diff
@@ -138,28 +137,18 @@ func (gc *GoCommand) run() error {
 	if err != nil {
 		return err
 	}
-
-	var goBuild *build.Build
-	toCollect, err := gc.buildConfiguration.IsCollectBuildInfo()
+	goBuildInfo, err := utils.PrepareBuildPrerequisites(gc.buildConfiguration)
 	if err != nil {
 		return err
 	}
-	if toCollect {
-		buildName, err := gc.buildConfiguration.GetBuildName()
-		if err != nil {
-			return err
-		}
-		buildNumber, err := gc.buildConfiguration.GetBuildNumber()
-		if err != nil {
-			return err
-		}
-		projectKey := gc.buildConfiguration.GetProject()
-		buildInfoService := utils.CreateBuildInfoService()
-		goBuild, err = buildInfoService.GetOrCreateBuildWithProject(buildName, buildNumber, projectKey)
-		if err != nil {
-			return errorutils.CheckError(err)
+	defer func() {
+		if goBuildInfo != nil && err != nil {
+			e := goBuildInfo.Clean()
+			if e != nil {
+				err = errors.New(err.Error() + "\n" + e.Error())
+			}
 		}
-	}
```

Comment:
I suggest implementing the following changes, to make the code straight forward.
1. Make the function return a named returned value, meaning ```(err error)``` instead of ```error```.
2. Make sure err isn't created throughout the function, but the named ```error``` is used.
3. Replace ```return err``` with ```return```.
4. Change the code inside the defer from - 
```
	defer func() {
		if goBuildInfo != nil && err != nil {
			e := goBuildInfo.Clean()
			if e != nil {
				err = errors.New(err.Error() + "\n" + e.Error())
			}
		}
	}
```
to
```
	defer func() {
		if goBuildInfo != nil {
			e := goBuildInfo.Clean()
			if err == nil {
				err = errorutils.CheckError(e)
			}
		}
	}
```
5. All the the above also applies to the ```defer``` clause in the ```Run()``` function in ```python.go```/